### PR TITLE
Add recipe for lib3mf

### PR DESCRIPTION
**Pull request by StephenSmith25**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by StephenSmith25** [comment]
Ready for review @conda-forge/help-c-cpp

File: recipes/lib3mf/build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -x
+
+mkdir build
+cd build
+cmake \
+    -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
+    -DCMAKE_INSTALL_LIBDIR="${PREFIX}/lib" \
```
Comment (xhochy) [inline recipes/lib3mf/build.sh:9]: Use CMAKE_ARGS for the most standard CMake arguments on conda-forge.
Also, please use zlib & co from conda-forge.
```suggestion
    ${CMAKE_ARGS} \
    -DUSE_INCLUDED_ZLIB=OFF -DUSE_INCLUDED_LIBZIP=OFF -DUSE_INCLUDED_SSL=OFF \
```

File: recipes/lib3mf/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    - if not exist %PREFIX%\\Library\\lib\\lib3mf.lib exit 1                           # [win]
+
+about:
+  home: https://github.com/3MFConsortium/lib3mf
+  license: BSD-2-Clause
+  license_family: BSD
+  license_file: 
+    - LICENSE
+    - library_licenses/libzip-LICENSE-BSD-3
+    - library_licenses/zlib-LICENSE-ZLIB
```
Comment (xhochy) [inline recipes/lib3mf/meta.yaml:40]: If you use the conda-forge libs, you don't need to include these two
```suggestion
```

File: recipes/lib3mf/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler("cxx") }}
+    - cmake
+    - pkg-config  # [unix]
+    - ninja  # [unix]
```
Comment (xhochy) [inline recipes/lib3mf/meta.yaml:22]:
```suggestion
    - ninja  # [unix]
  host:
    - libzip
    - zlib
```

**Review by xhochy — CHANGES_REQUESTED** [review]
Please use the conda-forge third-party library. The CMakeLists of the project already seems to support this.

File: recipes/lib3mf/build.sh (hunk 1 above)
Comment (StephenSmith25) [reply recipes/lib3mf/build.sh]: Should I use %CMAKE_ARGS% in bld.bat?

**Comment by StephenSmith25** [comment]
Thanks for looking at that and finding out I could use zlib ect. from conda-forge. I've made the changes you requested, but just one question about using %CMAKE_ARGS% in the bld.bat file.

File: recipes/lib3mf/build.sh (hunk 1 above)
Comment (xhochy) [reply recipes/lib3mf/build.sh]: No, we don't have that for Windows yet.

**Comment by xhochy** [comment]
> Thanks for looking at that and finding out I could use zlib ect. from conda-forge. I've made the changes you requested, just one question about using %CMAKE_ARGS% in the bld.bat file. Seems its having trouble with pkg config on windows trying to locate `zlib` and `libzip`, cant say I'm very familiar with how `pkg config` works. Will this be possible or will I need to patch the cmake file?

I'm also no expert on the build issues on Windows. If it makes trouble, feel free to statically link there for now.

**Comment by StephenSmith25** [comment]
I can't seem to get it working locally on windows using the `-DUSE_INCLUDED_ZLIB=OFF -DUSE_INCLUDED_LIBZIP=OFF -DUSE_INCLUDED_SSL=OFF` flags. Would it be okay to use the internally included libraries for now? If so, should I still use the Conda forge libraries on OSX and Linux, or use the internal ones across all three OSs?

**Comment by xhochy** [comment]
Use the bundled ones on Win, the conda-forge ones on Linux and OSX.

**Review by xhochy — APPROVED** [review]

**Pull request merged**